### PR TITLE
Nexus: Add rotate and matrix_transform functions 

### PR DESCRIPTION
**Pull request by mcbennet**
This PR adds the function `rotate` to `structure` and allows the user to perform arbitrary rotations of the structure. Additionally, the function `matrix_transform` is added which allows the user to transform the structure in an arbitrary way (matrix need not be orthogonal).

Both `rotate` and `matrix_transform`, when given explicit matrices, are assumed to be in column-major form, i.e., they transform a vector v as v'=Tv

### CODE REVIEW

File: nexus/lib/structure.py [hunk 1]
```diff
@@ -1457,21 +1551,149 @@ def stretch(self,s1,s2,s3):
...
-        kaxnew = dot(self.kaxes,inv(skew).T)
+        kaxnew = dot(self.kaxes,inv(A).T)
         self.pos     = dot(dot(self.pos,axinv),axnew)
         self.center  = dot(dot(self.center,axinv),axnew)
         self.kpoints = dot(dot(self.kpoints,kaxinv),kaxnew)
         self.axes  = axnew
         self.kaxes = kaxnew
         if self.folded_structure!=None:
-            self.folded_structure.skew(skew)
+            self.folded_structure.matrix_transform(A)
```
Comment (jtkrogel) [inline nexus/lib/structure.py]: Should this be `self.folded_structure.matrix_transform(A.T)` instead?

Comment (mcbennet) [reply nexus/lib/structure.py]: Yes, very good.

**Comment by markdewing** [comment]
okay to test

**Comment by markdewing** [comment]
@jtkrogel Are you okay with this PR?

**Comment by jtkrogel** [comment]
Yes.

**Pull request merged**